src/Microsoft.AspNetCore.Mvc.Razor.Precompilation.Design/Internal/PrecompilationApplication.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.Extensions.CommandLineUtils;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Precompilation.Design.Internal
+{
+    public class PrecompilationApplication : CommandLineApplication
+    {
+        private readonly Type _callingType;
+
+        public PrecompilationApplication(Type callingType)
+        {
+            _callingType = callingType;
+
+            Name = "razor-precompile";
+            FullName = "Microsoft Razor Precompilation Utility";
+            Description = "Precompiles Razor views.";
+            ShortVersionGetter = GetInformationalVersion;
+
+            HelpOption("-?|-h|--help");
+
+            OnExecute(() =>
+            {
+                ShowHelp();
+                return 2;
+            });
+        }
+
+        public new int Execute(params string[] args)
+        {
+            try
+            {
+                return base.Execute(args);
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine(ex.Message);
```

Comment:
Why hide the stack trace? If a customer reports a bug that's going to be the first thing we ask them for.